src/modules/sync/lib/ipc-handlers.ts

```diff
@@ -158,16 +161,99 @@ export async function pushArchive(
 		throw new Error( 'No token found' );
 	}
 
+	let hasUploadStarted = false;
+	let isUploadingPaused = false;
+	const file = fs.createReadStream( archivePath );
+	const fileSize = fs.statSync( archivePath ).size;
+	const filename = path.basename( archivePath );
+
+	const attachmentPromise = new Promise< string >( ( resolve, reject ) => {
+		const upload = new Upload( file, {
+			endpoint: `https://public-api.wordpress.com/rest/v1.1/studio-file-uploads/${ remoteSiteId }`,
+			chunkSize: 500000,
+			retryDelays: [ 0, 1000, 3000, 5000, 10000, 25000 ],
+			overridePatchMethod: true,
+			removeFingerprintOnSuccess: true,
+			storeFingerprintForResuming: true,
+			headers: {
+				Authorization: `Bearer ${ token.accessToken }`,
+			},
+			metadata: {
+				filename,
+				filetype: 'application/gzip',
+			},
+			uploadSize: fileSize,
+			onBeforeRequest: ( req ) => {
+				if ( req.getMethod() === 'HEAD' ) {
+					// @ts-expect-error We need to override the method to get the response headers.
+					req._method = 'GET';
+					req.setHeader( 'X-HTTP-Method-Override', 'HEAD' );
+				}
+			},
+			onError: ( error ) => {
+				console.error( '[TUS] Upload error', error );
+				reject( error );
+			},
+			onProgress: ( bytesSent, bytesTotal ) => {
+				if ( isUploadingPaused ) {
+					isUploadingPaused = false;
+					void sendIpcEventToRenderer( 'sync-upload-resumed', {
+						selectedSiteId: selectedSiteId,
+						remoteSiteId: remoteSiteId,
+					} );
+					console.log( '[TUS] Upload resumed' );
+				}
+
+				if ( ! hasUploadStarted ) {
+					hasUploadStarted = true;
+				}
+
+				if ( bytesTotal > 0 ) {
+					const progress = ( bytesSent / bytesTotal ) * 100;
+					console.log( '[TUS] Upload progress: %s%%', progress.toFixed( 2 ) );
+				}
+			},
+			onSuccess: ( payload ) => {
+				if ( ! payload.lastResponse ) {
+					console.error( 'Upload completed but no response received' );
+					reject( new Error( 'Upload completed but no response received' ) );
+					return;
+				}
+
+				const attachmentId = payload.lastResponse.getHeader( 'x-videopress-upload-media-id' );
+				if ( attachmentId ) {
+					resolve( attachmentId );
+				} else {
+					console.error( 'Upload completed but required header not found' );
+					reject( new Error( 'Upload completed but required header not found' ) );
+				}
+			},
+			onShouldRetry: ( error ) => {
+				// Update the UI only if the upload has started and is paused for any reason.
+				if ( hasUploadStarted ) {
+					isUploadingPaused = true;
+					void sendIpcEventToRenderer( 'sync-upload-paused', {
+						selectedSiteId: selectedSiteId,
+						remoteSiteId: remoteSiteId,
+						error: error.message,
+					} );
+				}
+
+				console.error( 'Upload request error', error.message );
```

Comment:
Should we log `Upload request error` here, or something that makes it clearer that the upload is now paused?